### ISSUE TITLE
[JENKINS-28595] Tests execution stops in a deadlock

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#9814]
<p>When I try to run the tests the result is a hanged process .</p>

<p>It seems to always stop in this test:</p>

<pre>
Running hudson.tools.ToolLocationTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.004 sec - in hudson.tools.ToolLocationTest
</pre>

<p>You find attached information related to the process using jstack (output.txt)</p>

<p>This <a href="https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/2677/" class="external-link" target="_blank" rel="nofollow noopener">PR</a> can not be merged because it is blocked by this issue. Its execution ends with a timeout.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/recena.png" title="recena's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recena">recena</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-28595" target="_blank">Tests execution stops in a deadlock</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/recena.png" title="recena's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recena">recena</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2015-11-15T15:43:56+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When I try to run the tests the result is a hanged process .</p>

<p>It seems to always stop in this test:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Running hudson.tools.ToolLocationTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.004 sec - in hudson.tools.ToolLocationTest
</pre>
</div></div>

<p>You find attached information related to the process using jstack (output.txt)</p>

<p>This <a href="https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/2677/" class="external-link" target="_blank" rel="nofollow noopener">PR</a> can not be merged because it is blocked by this issue. Its execution ends with a timeout.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Java(TM) SE Runtime Environment (build 1.7.0_71-b14)<br/>
Ubuntu 14.04.2 LTS (64bits)<br/>
Apache Maven 3.2.5<br/>
Jenkins (MASTER branch)
```
</details>
<details><summary><i>1 attachment</i></summary>

- [output.txt](https://issues.jenkins.io/secure/attachment/29851/output.txt)
</details>
